internal/ir/infra.go

```diff
@@ -0,0 +1,65 @@
+package ir
+
+import (
+	cfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
+// Infra defines managed infrastructure.
+type Infra struct {
+	// Proxy defines managed proxy infrastructure.
+	Proxy *ProxyInfra
+}
+
+// ProxyInfra defines managed proxy infrastructure.
+type ProxyInfra struct {
+	// TODO: Figure out how to represent metadata in the IR.
+	// xref: https://github.com/envoyproxy/gateway/issues/173
+	//
+	// Name is the name used for managed proxy infrastructure.
+	Name string
+	// Namespace is the namespace used for managed proxy infrastructure.
+	Namespace string
+	// OwnerRefs is a list of objects used to generate this object.
+	OwnerRefs []ObjectReference
+	// Config defines user-facing configuration of the managed proxy infrastructure.
+	Config *cfgv1a1.EnvoyProxy
+	// CertificateRefs is a list of object references that contain TLS certificates
+	// and private keys. These certificates are used to establish secure communication
+	// between the control and data plane.
+	CertificateRefs []LocalObjectReference
```

Comment:
this assumes that there will be a secret created before hand, what about the case where EG needed to create certs
for control plane authn. Since this is yet to be designed (https://github.com/envoyproxy/gateway/issues/70) suggest leaving this out for now and bringing it in as part of the above issue

Reply:
> this assumes that there will be a secret created before hand, what about the case where EG needed to create certs
for control plane authn. 

EG will create the cert/key for control plane auth and store them in a secret. The `ProxyInfra` will reference the cert/key created by EG.

Reply:
👍 prefer adding this field when we have decided on authn design

Reply:
If we don't have this field present to start with, we can't test what happens with preprovisioned certs (which is the easier case, and worth doing first).